utils/chart-lint.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func main() {
+	// recursively find Chart.yaml files
+	charts, err := findChartDirs(".")
+	if err != nil {
+		fmt.Println("Error:", err)
+		log.Fatal(err)
+	}
+
+	fmt.Println("Found", len(charts), "charts")
+
+	// convert chart paths to absolute paths
+	for i, chart := range charts {
+		absPath, err := filepath.Abs(chart)
+		if err != nil {
+			fmt.Println("Error:", err)
+			log.Fatal(err)
+		}
+		charts[i] = absPath
+	}
+
+	chartErrors := 0
+
+	// for each Chart.yaml file, run helm lint
+	for _, chart := range charts {
+		fmt.Println("Linting", chart)
+
+		// cd to chart dir
+		err := os.Chdir(chart)
+		if err != nil {
+			fmt.Println("Error:", err)
+			log.Fatal(err)
+		}
+
+		// run helm lint
+		cmd := exec.Command("helm", "lint")
```

Comment:
I was trying to avoid creating  go mod related files by using only built-ins as there's going to be several simple scripts.  Maybe that's false economy?

Reply:
As mentioned, leaving it up to the author. The remark on the `go.mod` somewhat makes sense (to make each script as lightweight as possible). I personally prefer "portability" instead of "weight" in the case of these script. By embedding it, we're moving the "runtime" dependency (and weight) from the user to the developer, but it does make the script very easy to put into docker containers/CI.

And I've seen "just use the CLI binary" go wrong way too many times, although in this case it's pretty simple.

Pragmatically, I'd accept the solution for now (to get the work done), make mental note of it and if it triggers bugs or when it's a slow day :sweat_smile: do it "my way"